archai/nlp/nvidia_transformer_xl/mem_transformer.py

```diff
@@ -811,7 +811,7 @@ def _forward(self, dec_inp, mems=None, past_key_values=None):
         plen = past_key_values[0][0].size(0) if past_key_values[0] is not None else 0
         klen = mlen + qlen
         if self.same_length:
-            all_ones = word_emb.new_ones(qlen, klen)
+            all_ones = word_emb.new_ones(qlen, klen+plen)
```

Comment:
Could you please add comment on why this change was done?

Reply:
I have added a new commit with the following comment on why we need to add `plen`:

"`plen` should be taken into account when creating the attention mask because `past_key_values` might be used."